### PR TITLE
feat: added doc for self-hosting the mtls ambassador

### PR DESCRIPTION
**Pull request by oldgiova**
Ticket: MC-6908

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure that all commits follow the conventional commit [specification](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: adding self-hosted mTLS Ambassador doc

Changelog: added self-hosted mTLS Ambassador doc
Ticket: MC-6908
```

- [x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Added self-hosted mTLS Ambassador doc

### CODE REVIEW

File: 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  ```bash
+  kubectl create secret generic keycert --from-file=cert.crt=./cert.crt --from-file=private.key=./private.key
+  ```
+* Create the CA secret:
+  ```bash
+  kubectl create secret generic mtls-ca --from-file=ca.crt=./ca.crt
+  ```
+* Create the registry secret:
+  ```bash
+  kubectl create secret registry-mtls-secret regcred --docker-server=registry.mender.io --docker-username=<your-name> --docker-password=<your-pword>
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md:41]: This didn't work for me on a local k3s.

```
$ kubectl create secret registry-mtls-secret regcred --docker-server=registry.me
nder.io --docker-username=12345 --docker-password=12345
error: unknown flag: --docker-server
See 'kubectl create secret --help' for usage.
```

Had to change it to:
```
kubectl create secret docker-registry regcred --docker-server=registry.mender.io --docker-username=12345 --docker-password=12345
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+         type: Utilization
+         averageUtilization: ${SCALABILITY_AVERAGE_CPU_UTILIZATION}
+ 
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: mender-mtls
+EOF
+
+kubectl apply -f mender-mtls-deployment.yml
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md:142]: This failed (again on k3s on my local machine):

```
t$ kubectl apply -f mender-mtls-deployment.
yml
horizontalpodautoscaler.autoscaling/mender-mtls unchanged
Error from server (BadRequest): error when creating "mender-mtls-deployment.yml": Deployment in version "
v1" cannot be handled as a Deployment: strict decoding error: unknown field "spec.template.metadata.conta
iners", unknown field "spec.template.metadata.imagePullSecrets", unknown field "spec.template.metadata.vo
lumes"
```

**Review by TheMeaningfulEngineer — CHANGES_REQUESTED** [review]
Made a PR for your PR to fix the formating: https://github.com/oldgiova/mender-docs/pull/1   
Going forward it might be simpler that you give be push access to your branch.

For the next iteration would explore options to run this on a real AWS or Azure cluster.
Couldn't get it to run this way.

**Review by TheMeaningfulEngineer — CHANGES_REQUESTED** [review]
I've tried this again with k3s (set up a separate PC as a cluster).
But it's failing with:

```
horizontalpodautoscaler.autoscaling/mender-mtls unchanged
Error from server (BadRequest): error when creating "mender-mtls-deployment.yml": Deployment in version "v1" cannot be handled a
s a Deployment: strict decoding error: unknown field "spec.template.metadata.containers", unknown field "spec.template.metadata.
imagePullSecrets", unknown field "spec.template.metadata.volumes"
```

File: 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md (hunk 1 above)
Comment (oldgiova) [reply 07.Server-installation/04.Production-installation-with-kubernetes/04.mTLS-Ambassador/docs.md]: Solved

**Comment by oldgiova** [comment]
> Made a PR for your PR to fix the formating: [oldgiova#1](https://github.com/oldgiova/mender-docs/pull/1) Going forward it might be simpler that you give be push access to your branch.
> 
> For the next iteration would explore options to run this on a real AWS or Azure cluster. Couldn't get it to run this way.

Just tried on a real EKS cluster:
```
▶ kk get pods,hpa,secret,svc            
NAME                                           READY   STATUS    RESTARTS   AGE
pod/cert-manager-776649d6c6-6zdjb              1/1     Running   0          16m
pod/cert-manager-cainjector-7bb8cb69c5-dcp49   1/1     Running   0          36m
pod/cert-manager-webhook-5c8bfb9bdf-ss8rp      1/1     Running   0          18m
pod/mender-mtls-64d88d4644-45d7r               1/1     Running   0          4m20s

NAME                                              REFERENCE                TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
horizontalpodautoscaler.autoscaling/mender-mtls   Deployment/mender-mtls   <unknown>/70%   1         10        1          17m

NAME                                        TYPE                             DATA   AGE
secret/cert-manager-webhook-ca              Opaque                           3      36m
secret/keycert                              Opaque                           2      31m
secret/mtls-ca                              Opaque                           1      30m
secret/mtls-user                            Opaque                           2      32m
secret/registry-mtls-secret                 kubernetes.io/dockerconfigjson   1      27m
secret/sh.helm.release.v1.cert-manager.v1   helm.sh/release.v1               1      36m

NAME                           TYPE           CLUSTER-IP       EXTERNAL-IP                                                                       PORT(S)         AGE
service/cert-manager           ClusterIP      10.100.235.66    <none>                                                                            9402/TCP        36m
service/cert-manager-webhook   ClusterIP      10.100.121.29    <none>                                                                            443/TCP         36m
service/kubernetes             ClusterIP      10.100.0.1       <none>                                                                            443/TCP         48m
service/mender-mtls            LoadBalancer   10.100.213.193   k8s-default-mendermt-4783d9343d-6acd60919d907237.elb.eu-central-1.amazonaws.com   443:30355/TCP   2m38s
```

**Comment by oldgiova** [comment]
> I've tried this again with k3s (set up a separate PC as a cluster). But it's failing with:
> 
> ```
> horizontalpodautoscaler.autoscaling/mender-mtls unchanged
> Error from server (BadRequest): error when creating "mender-mtls-deployment.yml": Deployment in version "v1" cannot be handled a
> s a Deployment: strict decoding error: unknown field "spec.template.metadata.containers", unknown field "spec.template.metadata.
> imagePullSecrets", unknown field "spec.template.metadata.volumes"
> ```

Solved, thanks!

**Comment by TheMeaningfulEngineer** [comment]
@oldgiova 
I'd like to close this PR in favor of [this](https://github.com/mendersoftware/mender-docs/pull/2165).

We're redoing this whole section on mtls and would like to have the mtls kubernetes tutorial as part of it.
I've copied the whole content from here into the other PR and would like to close this one.

Here is a render of how it is planned:
Screenshot from 2023-09-27 14-08-22
 
 What could change for you with this - instead of reviewing the content here if we find bugs, you'd have to review it a different PR. It will have much more content, but the kubernetes installation will remain an independent doc file with the same structure as here.
 
 Does this work for you?

**Comment by oldgiova** [comment]
> Does this work for you?

Yes, it does!

**Pull request closed without merging**